File: tntrun/submissions/accepted/tntrun_ac.cpp
#include <iostream>
#include <vector>
#include <cassert>

using namespace std;

/**
 * Output YES if it is possible for the player to run on the course such that it
 * consists of blocks and banned otherwise.
 * 
 * N: the number of blocks in the course.
 * S: the list of starting blocks
 * E: the list of ending blocks
 */
void solve(int N, vector<char> S, vector<char> E) {
    bool ans = true;
    int cnt = 0;
    for (int i = 0; i < N; i++) {
        if (S[i] == '-' && E[i] == '#') { // Added a TNT block
            assert(false); // This should happen because of test case constraints
            ans = false;
        } else if (S[i] == '#' && E[i] == '-') { // Used a TNT block to jump
            cnt = 0;
        } else { // Jumped over this block
            ++cnt;
            if (cnt > 4) {
                ans = false;
            }
        }
    }

    if (ans) {
        cout << "YES\n";
    } else {
        cout << "banned\n";
    }
    
    return;
}

int main() {
    int T;
    cin >> T;
    for (int i = 0; i < T; i++) {
        int N;
        cin >> N;
        vector<char> S(N);
        for (char &item : S) {
            cin >> item;
        }
        vector<char> E(N);
        for (char &item : E) {
            cin >> item;
        }
        solve(N, S, E);
    }
}

File: tntrun/submissions/accepted/tntrun_ac.java
import java.io.*;

class Solution {
    /**
     * Output YES if it is possible for the player to run on the course such that it
     * consists of blocks and banned otherwise.
     * 
     * N: the number of blocks in the course.
     * S: the list of starting blocks
     * E: the list of ending blocks
     */
    static void solve(int N, char[] S, char[] E) {
        boolean ans = true;
        int cnt = 0;
        for (int i = 0; i < N; ++i) {
            if (S[i] == '-' && E[i] == '#') {
                ans = false;
            } else if (S[i] == '#' && E[i] == '-') {
                cnt = 0;
            } else {
                ++cnt;
                if (cnt > 4) {
                    ans = false;
                }
            }
        }
        if (ans) {
            out.println("YES");
        } else {
            out.println("banned");
        }
        return;
    }
    
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static PrintWriter out = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        int T = Integer.parseInt(in.readLine());
        for (int i = 0; i < T; i++) {
            int N = Integer.parseInt(in.readLine());
            char[] S = new char[N];
            String temp = in.readLine();
            for (int j = 0; j < N; j++) {
                S[j] = temp.charAt(j);
            }
            char[] E = new char[N];
            temp = in.readLine();
            for (int j = 0; j < N; j++) {
                E[j] = temp.charAt(j);
            }
            solve(N, S, E);
        }
        out.flush();
    }
}
